consensus/core/src/transaction.rs

```diff
@@ -720,4 +722,86 @@ mod tests {
             assert!(r.is_ok());
         }
     }
+
+    #[tokio::test]
+    async fn test_submit_over_max_block_size_and_validate_block_size() {
+        let _guard = ProtocolConfig::apply_overrides_for_testing(|_, mut config| {
+            config.set_consensus_max_transaction_size_bytes_for_testing(100);
+            config.set_consensus_max_num_transactions_in_block_for_testing(10);
+            config.set_consensus_max_transactions_in_block_bytes_for_testing(300);
+            config
+        });
+
+        let context = Arc::new(Context::new_for_test(4).0);
+        let (client, tx_receiver) = TransactionClient::new(context.clone());
+        let mut consumer = TransactionConsumer::new(tx_receiver, context.clone());
+        let mut all_receivers = Vec::new();
+
+        // submit transactions individually so we make sure that we have reached the block size limit of 10
+        {
+            // create enough transactions
+            let max_num_transactions_in_block =
+                context.protocol_config.max_num_transactions_in_block();
+            for i in 0..2 * max_num_transactions_in_block {
+                let transaction = bcs::to_bytes(&format!("transaction {i}"))
+                    .expect("Serialization should not fail.");
+                let w = client
+                    .submit_no_wait(vec![transaction])
+                    .await
+                    .expect("Should submit successfully transaction");
+                all_receivers.push(w);
+            }
+
+            // Fetch the next transactions to be included in a block
+            let (transactions, _ack_transactions) = consumer.next();
+            assert!(transactions.len() as u64 == max_num_transactions_in_block);
+
+            // Now create a block and verify that transactions are within the size limits
+            let block_verifier =
+                SignedBlockVerifier::new(context.clone(), Arc::new(NoopTransactionVerifier {}));
+
+            let batch: Vec<_> = transactions.iter().map(|t| t.data()).collect();
+            assert!(
+                block_verifier.check_transactions(&batch).is_ok(),
+                "Number of transactions limit verification failed"
+            );
+        }
+
+        // submit transactions individually so we make sure that we have reached the block size bytes 300
+        {
+            let max_transactions_in_block_bytes =
+                context.protocol_config.max_transactions_in_block_bytes();
+            let mut total_size = 0;
+            loop {
+                let transaction = bcs::to_bytes(&"transaction".to_string())
```

Comment:
Is this actually testing hitting the size limits? Maybe use 50-byte payloads instead?

Reply:
@mwtian did you still want to update the test?

Reply:
With these, using smaller transactions are ok:
```
config.set_consensus_max_num_transactions_in_block_for_testing(1_000);
config.set_consensus_max_transactions_in_block_bytes_for_testing(300);
```